app/Http/Controllers/HomeController.php

```diff
@@ -25,4 +26,97 @@ public function index()
     {
         return view('admin.home');
     }
+
+    /**
+     * Show the user cart
+     *
+     * @return \Illuminate\Contracts\Support\Renderable
+     */
+    public function viewCart()
+    {
+        return view('client.cart.index');
+    }
+
+    /**
+     * Add Product to Cart.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function addProductToCart($id)
+    {    
+        $product = Product::findOrFail($id);        
+        $cart = session()->get('cart');
+
+        // if cart is empty then this the first product
+        if (!$cart) {
+            $cart = [
+                $id => [
+                    "name" => $product->name,
+                    "quantity" => 1,
+                    "price" => $product->price,
+                    "featured_img" => $product->featured_img,                        
+                ],
+            ];
+            session()->put('cart', $cart);
+
+            return redirect()->back()->with('success', __('add_product_successfully'));
+        }
+        // if cart not empty then check if this product exist then increase quantity
+        if (isset($cart[$id])) {
+            $cart[$id]['quantity']++;
+            session()->put('cart', $cart);
+
+            return redirect()->back()->with('success', __('add_product_successfully'));
+        }
+        // if item not exist in cart then add to cart with quantity = 1
+        $cart[$id] = [
+            "name" => $product->name,
+            "quantity" => 1,
+            "price" => $product->price,
+            "featured_img" => $product->featured_img,            
+        ];
+        session()->put('cart', $cart);
+
+        return redirect()->back()->with('success', __('add_product_successfully'));
+    }
+
+    /**
+     * Update Cart.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function updateCart(Request $request)
+    {            
+        if($request->id && $request->quantity)
+        {
+            $cart = session()->get('cart');
+            $cart[$request->id]["quantity"] = $request->quantity;
```

Comment:
```suggestion
            $cart[$request->id]['quantity'] = $request->quantity;
```
dấu " và dấu ' có gì khác nhau?

Reply:
- Dấu nháy đơn ('): Chuỗi trong dấu nháy đơn không được phân tích cú pháp, có nghĩa là những gì được viết trong dấu nháy đơn sẽ được xuất hiện. Các ký ký hiệu \n (xuống dòng), \t (tab) sẽ không được xác định.
- Dấu nháy kép ("): Chuỗi trong dấu nháy kép được phấn tích cú pháp và bất kỳ biến trong php sẽ được xác định. Các ký ký hiệu \n (xuống dòng), \t (tab) được xác định.
Vì chuỗi trong dấu nháy kép được phân tích trong quá trình thực thi, theo lý thuyết thì sử dụng dấu nháy đơn sẽ cải thiện hiệu suất vì trong PHP không phải đánh giá chuổi đơn. Điều này đúng với một mô hình nhất định, với ứng dụng trong thực tế nếu ở mức độ trung bình thì sự khác biệt là quá nhỏ nên điều này không thực sự quan trọng. Vì vậy đối với úng dụng trung bình nó không có vấn đề gì để bạn phải lựa chọn. Nếu với ứng dụng tải rất cao thì chúng ta nên xem xét lựa chọn.